### PR TITLE
Updates from testing

### PR DESCRIPTION
**Pull request by charles-cowart**
The following updates were made to klp.py during testing by Antonio and Charlie.

### CODE REVIEW

File: .github/workflows/qiita-plugin-ci.yml [hunk 1]
```diff
@@ -89,7 +90,7 @@ jobs:
           pip --quiet install coveralls
```
Comment (antgonza) [inline .github/workflows/qiita-plugin-ci.yml:91]:
```suggestion
          export QP_KLP_CONFIG_FP='NOTHING'

```

File: .github/workflows/qiita-plugin-ci.yml [hunk 2]
```diff
@@ -87,9 +88,10 @@ jobs:
           pip --quiet install https://github.com/qiita-spots/qtp-job-output-folder/archive/refs/heads/main.zip
           pip --quiet install .
           pip --quiet install coveralls
+          export QP_KLP_CONFIG_FP='NOTHING'
```
Comment (antgonza) [inline .github/workflows/qiita-plugin-ci.yml:91]:
```suggestion
          export QP_KLP_CONFIG_FP=`pwd`/configuration.json
```

File: .github/workflows/qiita-plugin-ci.yml [hunk 3]
```diff
@@ -87,9 +88,10 @@ jobs:
...
 
           configure_qtp_job_output_folder --env-script "source /home/runner/.profile; conda activate klp" --server-cert $QIITA_SERVER_CERT
-          configure_klp --env-script "source /home/runner/.profile; conda activate klp" --server-cert $QIITA_SERVER_CERT
+          configure_klp --env-script "source /home/runner/.profile; export QP_KLP_CONFIG_FP='NOTHING'; conda activate klp" --server-cert $QIITA_SERVER_CERT
```
Comment (antgonza) [inline .github/workflows/qiita-plugin-ci.yml:94]:
```suggestion
          configure_klp --env-script "source /home/runner/.profile; export QP_KLP_CONFIG_FP=$QP_KLP_CONFIG_FP; conda activate klp" --server-cert $QIITA_SERVER_CERT
```

File: .github/workflows/qiita-plugin-ci.yml [hunk 4]
```diff
@@ -132,7 +134,7 @@ jobs:
           conda activate klp
           export QIITA_SERVER_CERT=`pwd`/qiita-dev/qiita_core/support_files/server.crt
           export QIITA_CONFIG_FP=`pwd`/qiita-dev/qiita_core/support_files/config_test_local.cfg
-
+          export QP_KLP_CONFIG_FP='NOTHING'
```
Comment (antgonza) [inline .github/workflows/qiita-plugin-ci.yml:137]:
```suggestion
          export QP_KLP_CONFIG_FP=`pwd`/configuration.json
```

File: qp_klp/tests/test_klp.py [hunk 5]
```diff
@@ -5,32 +5,230 @@
...
+              }
+            }
+
+        # use out_dir to store the configuration.json file as well.
+        # create the file and write the configuration out to disk
+        # for use by sequence_processing_pipeline().
+        config_filepath = join(self.out_dir, 'configuration.json')
+        CONFIG_FP = config_filepath
+        environ['QP_KLP_CONFIG_FP'] = config_filepath
+        print(CONFIG_FP)
```
Comment (antgonza) [inline qp_klp/tests/test_klp.py:221]:
```suggestion
        # use out_dir to store the configuration.json file as well.
        # create the file and write the configuration out to disk
        # for use by sequence_processing_pipeline().
        config_filepath = environ['QP_KLP_CONFIG_FP']
```

File: qp_klp/tests/test_klp.py [hunk 6]
```diff
@@ -5,32 +5,230 @@
...
+
+        with open(config_filepath, 'w') as f:
+            f.write(dumps(spp_config, indent=2))
+
+        # Set parameters for qp_klp.py:
+        #  config-fp is the file path to qp-klp's configuration.json file.
+        #  skip-exec informs qp-klp to create objects and test parameters,
+        #   but don't execute actual jobs and processes.
+        environ["QP_KLP_CONFIG_FP"] = config_filepath
+
```
Comment (antgonza) [inline qp_klp/tests/test_klp.py:231]:
```suggestion

```

File: qp_klp/klp.py [hunk 7]
```diff
@@ -32,32 +44,228 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
...
+
+    # checking if this is running as part of the unittest
+    # https://stackoverflow.com/a/25025987
+    skip_exec = True if [x for x in stack() if
+                         'unittest' in x.filename] else False
 
     success = True
     ainfo = None
     msg = None
-    if exists(run_identifier) and isdir(run_identifier):
```
Comment (antgonza) [inline qp_klp/klp.py:42]: is this really no longer required? why?

File: qp_klp/klp.py [hunk 8]
```diff
@@ -32,32 +44,228 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
...
+            lst = bioinformatics.to_dict('records')
+
+        # find the uploads directory all trimmed files will need to be
+        # moved to.
+        results = qclient.get("/qiita_db/artifacts/types/")
+
+        # trimmed files are stored by qiita_id. Find the qiita_id
+        # associated with each project and ensure a subdirectory exists
+        # for when it comes time to move the trimmed files.
+        special_map = []
```
Comment (antgonza) [inline qp_klp/klp.py:92]: Do we really need to do this here or can it be done in line 235?

File: qp_klp/klp.py [hunk 9]
```diff
@@ -32,32 +44,228 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
...
+        for result in lst:
+            project_name = result['Sample_Project']
+            qiita_id = result['QiitaID']
+            upload_path = join(results['uploads'], qiita_id)
+            makedirs(upload_path, exist_ok=True)
+            special_map.append((project_name, upload_path))
+
+        config_fp = ('/home/qiita_test/qiita-spots/'
```
Comment (antgonza) [inline qp_klp/klp.py:100]: No more hardcoded paths please!!

**Review by antgonza — COMMENTED** [review]
some initial comments, will do a full pass later

File: qp_klp/klp.py (hunk 7 above)
Comment (charles-cowart) [reply qp_klp/klp.py:42]: It was appropriate before when run_identifier was being used to hold the full path to a run-dir, but the correct operation is to supply a run_id, and the Pipeline() object will determine the path based on its configured search paths. In this case, Pipeline() is already performing those two checks internally.

File: qp_klp/klp.py (hunk 8 above)
Comment (charles-cowart) [reply qp_klp/klp.py:92]: We want to get the list of valid project names from the sample-sheet. By doing it here, we're piggybacking on top of the necessary validation that needs to be performed. We don't want to keep a copy lst in memory the entire time, so it makes sense to at least extract the project names here.

File: qp_klp/klp.py (hunk 9 above)
Comment (charles-cowart) [reply qp_klp/klp.py]: Sorry, my bad. It's an old line that snuck back in during merging. It's been removed. Pipeline() should use the config_fp referred to in the actions file.

**Pull request merged**